manager/controllers/app/m4dapplication_controller.go

```diff
@@ -274,9 +287,51 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 	return ctrl.Result{}, nil
 }
 
-// ConstructDataInfo gets a list of governance actions and data location details for the given dataset and fills the received DataInfo structure
+// ConstructIngestDataInfo already has the information about the external data set's endpoint,
+// but it adds the governance decisions. It also already has the link to the credentials, which was provided by the user.
+// NOTE: Assumes the credentials were put into the M4D credential manager by the user!!
```

Comment:
Assuming credentials are already in vault, the appropriate module still needs to access them, i.e. the vault path should be provided to the module. See my comment in module_instance.go - management of credentials location should be different.

Reply:
Yes.  fixed